lib/ancestry/class_methods.rb

```diff
@@ -43,11 +43,11 @@ def arrange(options = {})
     # @returns Hash{Node => {Node => {}, Node => {}}}
     # If a node's parent is not included, the node will be included as if it is a top level node
     def arrange_nodes(nodes)
-      node_ids = Set.new(nodes.map(&:id))
+      node_ids = Set.new(nodes.map(&:ancestry_identifier_column))
       index = Hash.new { |h, k| h[k] = {} }
 
       nodes.each_with_object({}) do |node, arranged|
-        children = index[node.id]
+        children = index[node.ancestry_identifier_column]
```

Comment:
```suggestion
        children = index[node.send(ancestry_identifier_column)]
```

Reply:
`id` is nice. behind the scenes it does smart stuff. wonder if something similar would work here.

Reply:
Based on your [suggestion](https://github.com/stefankroes/ancestry/pull/698#discussion_r2098023509), I introduced the method `ancestry_identifier_value`, which returns `read_attribute(ancestry_identifier_column)`.

And `id` in Rails method implicitly returns the primary key value, so this `ancestry_identifier_value` can serve a similar purpose if I'm correct.

By using this method, I can avoid repeatedly writing `read_attribute` or `send` whenever I need to fetch the value, which helps keep the code more readable.